Source/pfile.cpp

```diff
@@ -250,89 +229,61 @@ unsigned char __fastcall game_2_ui_class(const PlayerStruct *p)
 
 BOOL __stdcall pfile_ui_set_hero_infos(BOOL (__stdcall *ui_add_hero_info)(_uiheroinfo *))
 {
-	char *v1; // esi
-	//int v2; // eax
-	int v3; // eax
-	DWORD v4; // eax
-	unsigned int v5; // esi
-	void *v7; // eax
-	void *v8; // edi
-	//int v9; // eax
-	bool v10; // al
-	PkPlayerStruct pkplr; // [esp+Ch] [ebp-7BCh]
-	struct _OFSTRUCT ReOpenBuff; // [esp+500h] [ebp-2C8h]
-	char FileName[260]; // [esp+588h] [ebp-240h]
-	char NewFileName[260]; // [esp+68Ch] [ebp-13Ch]
-	_uiheroinfo hero_info; // [esp+790h] [ebp-38h]
-	int unused; // [esp+7BCh] [ebp-Ch]
-	LPCSTR lpSrcStr; // [esp+7C0h] [ebp-8h]
-	int save_num; // [esp+7C4h] [ebp-4h]
+	unsigned int i, save_num;
+	char FileName[MAX_PATH];
+	char NewFileName[MAX_PATH];
+	int a1;
 
 	memset(hero_names, 0, sizeof(hero_names));
-	if ( (unsigned char)gbMaxPlayers > 1u )
-	{
-		lpSrcStr = 0;
-		save_num = 0;
-		do
-		{
-			if ( (unsigned int)save_num >= MAX_CHARACTERS )
-				break;
-			GetSaveDirectory(FileName, 260, (int)lpSrcStr);
-			v1 = strrchr(FileName, '\\') + 1;
-			if ( v1 != (char *)1 && OpenFile(FileName, &ReOpenBuff, 0x4000u) != -1 )
-			{
-				if ( !SRegLoadString("Diablo\\Converted", (const char *)v1, 0, NewFileName, 260) )
-				{
-					while ( 1 )
-					{
-						v3 = save_num++;
-						pfile_get_save_path(NewFileName, 260, v3);
-						if ( OpenFile(NewFileName, &ReOpenBuff, 0x4000u) == -1 )
-							break;
-						if ( (unsigned int)save_num >= MAX_CHARACTERS )
-							goto LABEL_13;
-					}
-					if ( CopyFile(FileName, NewFileName, 1) )
-					{
-						SRegSaveString("Diablo\\Converted", v1, 0, NewFileName);
-						v4 = GetFileAttributes(NewFileName);
-						if ( v4 != -1 )
-						{
-							_LOBYTE(v4) = v4 & 0xF9;
-							SetFileAttributes(NewFileName, v4);
+	if (gbMaxPlayers > 1) {
+		for ( i = 0, save_num = 0; i < MAX_CHARACTERS && save_num < MAX_CHARACTERS; i++) {
+			struct _OFSTRUCT ReOpenBuff;
+			const char *s;
+
+			GetSaveDirectory(FileName, sizeof(FileName), i);
+			s = strrchr(FileName, '\\') + 1;
+			if ( s == (const char*)1 )

```

Comment:
While this is the bin exact version. I would have preferred this to be written as:

```c
s = strrchr(FileName, '\\');
if ( s == NULL)
	continue; 
s++;
```

Oh, well. For recreation, lets keep it as the original.

Reply:
There's a lot of code I would prefer to write in different ways.